public/app/features/alerting/unified/mocks/server/mock-entities/alertmanagers.ts

```diff
@@ -0,0 +1,61 @@
+import alertmanagerConfigMock from 'app/features/alerting/unified/components/contact-points/__mocks__/alertmanager.config.mock.json';
+import { GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+import { AlertManagerCortexConfig, AlertmanagerStatus } from 'app/plugins/datasource/alertmanager/types';
+
+//////////////////////////
+// Alertmanager configs //
+//////////////////////////
+
+/** **INITIAL** state of alertmanager configs for different scenarios */
+const ALERTMANAGER_CONFIGS: Record<string, AlertManagerCortexConfig> = {
+  [GRAFANA_RULES_SOURCE_NAME]: alertmanagerConfigMock,
```

Comment:
This is causing a typescript error:

```
"message": "Type '{ template_files: { \"slack-template\": string; \"custom-email\": string; \"provisioned-template\": string; \"template with spaces\": string; \"misconfigured-template\": string; \"misconfigured and provisioned\": string; }; template_file_provenances: { ...; }; alertmanager_config: { ...; }; }' is not assignable to type 'AlertManagerCortexConfig'.
The types of 'alertmanager_config.route.routes' are incompatible between these types.
Type '({ match: { sub1matcher1: string; sub1matcher2: string; sub2matcher1?: undefined; sub2matcher2?: undefined; }; match_re: { sub1matcher3: string; sub1matcher4: string; sub2matcher3?: undefined; sub2matcher4?: undefined; }; ... 6 more ...; routes: ({ ...; } | { ...; })[]; } | { ...; } | { ...; })[]' is not assignable to type 'Route[]'.
Type '{ match: { sub1matcher1: string; sub1matcher2: string; sub2matcher1?: undefined; sub2matcher2?: undefined; }; match_re: { sub1matcher3: string; sub1matcher4: string; sub2matcher3?: undefined; sub2matcher4?: undefined; }; ... 6 more ...; routes: ({ ...; } | { ...; })[]; } | { ...; } | { ...; }' is not assignable to type 'Route'.
Type '{ match: { sub1matcher1: string; sub1matcher2: string; sub2matcher1?: undefined; sub2matcher2?: undefined; }; match_re: { sub1matcher3: string; sub1matcher4: string; sub2matcher3?: undefined; sub2matcher4?: undefined; }; ... 6 more ...; routes: ({ ...; } | { ...; })[]; }' is not assignable to type 'Route'.
Types of property 'match' are incompatible.
Type '{ sub1matcher1: string; sub1matcher2: string; sub2matcher1?: undefined; sub2matcher2?: undefined; }' is not assignable to type 'Record<string, string>'.
Property '\"sub2matcher1\"' is incompatible with index signature.
Type 'undefined' is not assignable to type 'string'.",
```

Because the shape of the `match`/`match_re` objects is different between each route in the mock config (https://github.com/grafana/grafana/blob/alerting/notif-policies-tests/public/app/features/alerting/unified/components/contact-points/__mocks__/alertmanager.config.mock.json#L34-L73)

The types define them as `Record<string, string>`, but it picks up the wrong properties. Any ideas? 🤷

Reply:
I wonder if we should keep these mock configs as json. Maybe it would be easier to keep them as typescript objects?